lib/neo4j/active_node/query/query_proxy.rb

```diff
@@ -186,11 +186,8 @@ def <<(other_node)
         #
         # @return [QueryProxy] A new QueryProxy
         def branch(&block)
-          if block
-            instance_eval(&block).query.proxy_as(self.model, identity)
-          else
-            fail LocalJumpError, 'no block given'
-          end
+          fail LocalJumpError, 'no block given' unless block
```

Comment:
I've ignored that rubocop rule in a number of projects, but I couldn't remember if I had done that here.  I like your `if block.nil?` suggestion from Gitter

I'm not a fanatic, exactly, but I have participated in conversations like this one: https://github.com/bbatsov/ruby-style-guide/issues/329